data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java

```diff
@@ -72,13 +75,13 @@ public static void main(final String[] args) {
     final var vertx = Vertx.vertx(
       new VertxOptions().setMetricsOptions(MetricsOptionsProvider.get(env, METRICS_REGISTRY_NAME))
     );
+    Runtime.getRuntime().addShutdownHook(new Thread(Shutdown.closeSync(vertx)));
```

Comment:
Why this? Just add one shutdown hook?

Reply:
If it fails before registering the other hook and after the verticle deploy from a readiness/liveness probe perspective the pod is ready, so closing Vertx makes sure we report not ready as soon as possible. WDYT?

Reply:
If something fails during the main, i think it doesn't make any difference having this hook or not.

Reply:
What happens if the application receives a SIGTERM with no hook registered?

Reply:
well, give it a try :smile: run locally the jar and check what happens

Reply:
I was looking here https://vertx-web-site.github.io/docs/vertx-core/java/#_causing_vert_x_to_exit so we need anyway to close vertx to close the application, i was wrong in the previous comment

Reply:
I've already tried and that is the reason for this PR. 
What happens is that the main thread finishes but the pod continues to be ready because the verticle server continues to respond.

Reply:
uhm, wrapping in the try catch all the "potentially failing" things after the creation of vertx makes sense?

My feeling is that adding 2 shutdown hooks that close vertx might be problematic

Reply:
When you add the second shutdown hook, you still have the first one mounted. So one will close vertx, the other won't do anything i think, because it's trying to close something already closed

Reply:
Calling close twice isn't problematic. I've checked the implementation.

> When you add the second shutdown hook, you still have the first one mounted. So one will close vertx, the other won't do anything i think, because it's trying to close something already closed

But the point is making sure we close Vertx in middle as well.

Reply:
Ok what happens if the first one you register is invoked before the second one? You won't have the shutdown as expected right?

Reply:
That's why I'm saying: try catch to wrap them all sounds reasonable and simpler to understand

Reply:
Ok, please check the last commit.